### PR TITLE
feat: mumu extras

### PR DESCRIPTION
**Pull request by MistEO**
- fix https://github.com/MaaAssistantArknights/MaaAssistantArknights/issues/8784

### CODE REVIEW

**Review by horror-proton — COMMENTED** [review]
应该不用自己搞引用计数吧

https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-freelibrary

> The reference count is decremented each time the FreeLibrary or FreeLibraryAndExitThread function is called for the module. When a module's reference count reaches zero or the process terminates, the system unloads the module from the address space of the process. Before unloading a library module, the system enables the module to detach from the process by calling the module's DllMain function, if it has one, with the DLL_PROCESS_DETACH value. Doing so gives the library module an opportunity to clean up resources allocated on behalf of the current process. After the entry-point function returns, the library module is removed from the address space of the current process.

https://man7.org/linux/man-pages/man3/dlopen.3.html

>       If the same shared object is opened again with dlopen(), the same
>       object handle is returned.  The dynamic linker maintains
>       reference counts for object handles, so a dynamically loaded
>       shared object is not deallocated until dlclose() has been called
>       on it as many times as dlopen() has succeeded on it.
>       Constructors (see below) are called only when the object is
>       actually loaded into memory (i.e., when the reference count
>       increases to 1).

**Comment by ABA2396** [comment]
Snipaste_2024-04-30_23-59-18
截图耗时11ms

File: src/MaaCore/AsstCaller.cpp [hunk 1]
```diff
@@ -163,15 +173,30 @@ AsstBool ASSTAPI AsstBackToHome(AsstHandle handle)
...
         return InvalidId;
     }
     return handle->async_connect(adb_path, address, config ? config : std::string(), block);
 }
 
+void AsstSetConnectionExtras(const char* name, const char* extras)
+{
+    auto jopt = json::parse(extras);
+    if (!jopt) {
+        LogError << "falied to parse json" << extras;
```
Comment (Constrat) [inline src/MaaCore/AsstCaller.cpp:193]: small typo

**Comment by Constrat** [comment]
The instance number and instance screen are related to these correct? 
image

**Comment by Rbqwow** [comment]
自动战斗最起飞的一局
测了一晚上发现了两个问题
1. 暂停下干员彻底用不了了 牛牛速度太快了 拖慢一点就好了
2. 截图延迟检测 除了刚启动第一次运行任务时检测正常几十毫秒 之后所有检测都会卡在几百毫秒不再变化 但实际上截图延迟很低

image

image

**Comment by Rbqwow** [comment]
文档写好了，看看 Maa 鼠标悬浮在输入框上需不需要也加点 tips。
TODO: 查明第一个可用的 MuMu 模拟器版本号，现在写的是最新版本号。
```
### MuMu 截图

需使用 MuMu 12 V3.8.21 (2869) 及更新版本，关闭后台保活。

#### 连接设置

尚未支持自动检测，下文中未提到的设置无需更改。

`连接配置` 选择 `MuMu 模拟器 12`。

`MuMu 模拟器路径` 填写 `MuMuPlayer-12.0` 文件夹的路径，如 `C:\Program Files\Netease\MuMuPlayer-12.0`。

`实例编号` 填写 MuMu 多开器内对应模拟器的序号，如主多开为 `0`。

`实例屏幕` 填 `0`。

#### 关于后台保活

推荐关闭，此时实例屏幕始终为 `0`。

开启时，MuMu 模拟器标签页的顺序即为实例屏幕的序号，如 `0` 为模拟器桌面，`1` 为明日方舟客户端。

针对后台保活的适配尚未完善，经常出现莫名其妙的问题，还是不建议使用。
```

**Comment by MistEO** [comment]
我感觉最好给 UI 显示时间那里在糊一个 “最快截图耗时：10ms （使用 MuMu 独家极速操控模式）"

**Comment by MistEO** [comment]
> The instance number and instance screen are related to these correct? image

Yes, this is instance number, and fill in 0 for the display number for now

**Comment by MistEO** [comment]
> 我感觉最好给 UI 显示时间那里在糊一个 “最快截图耗时：10ms （使用 MuMu 独家极速操控模式）"

另外，设置里是不是可以也提示一下这些是可选项，否则第一次用 MAA 还以为用 mumu 就必须要填这些，用别的模拟器就不用填，mumu 好麻烦（

**Pull request merged**